app2/ota_metadata.py

```diff
@@ -29,23 +46,67 @@ def __init__(self, ota_metadata_jwt):
         """
         self.__metadata_jwt = ota_metadata_jwt
         self.__metadata_dict = self._parse_metadata(ota_metadata_jwt)
+        self._certs_dir = OtaMetadata.CERTS_DIR
+
+    def verify(self, certificate: str):
+        """"""
+        # verify certificate itself before hand.
+        self._verify_certificate(certificate)
+
+        # verify metadata.jwt
+        try:
+            cert = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
+            verify_data = self._get_header_payload().encode()
```

Comment:
```suggestion
        # verify metadata.jwt
        try:
            # verify certificate itself before hand.
            self._verify_certificate(certificate)
            cert = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
            verify_data = self._get_header_payload().encode()
```

Reply:
Since _verify_certificate also raise exception, I think it is OK to put this line into the try, catch cluster?